src/crypto/CHIPCryptoPALmbedTLSCert.cpp

```diff
@@ -0,0 +1,1089 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      mbedTLS based implementation of CHIP crypto primitives related to certificate
+ *      generation and validation.
+ */
+
+#include "CHIPCryptoPAL.h"
+#include "CHIPCryptoPALmbedTLS.h"
+
+#include <lib/core/CHIPSafeCasts.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/SafeInt.h>
+
+#include <mbedtls/ecp.h>
+#include <mbedtls/oid.h>
+#include <mbedtls/version.h>
+#include <mbedtls/x509.h>
+#include <mbedtls/x509_csr.h>
+
+#if defined(MBEDTLS_X509_CRT_PARSE_C)
+#include <mbedtls/x509_crt.h>
+#endif // defined(MBEDTLS_X509_CRT_PARSE_C)
+
+namespace chip {
+namespace Crypto {
+
+// In mbedTLS 3.0.0 direct access to structure fields was replaced with using MBEDTLS_PRIVATE macro.
+#if (MBEDTLS_VERSION_NUMBER >= 0x03000000)
+#define CHIP_CRYPTO_PAL_PRIVATE(x) MBEDTLS_PRIVATE(x)
```

Comment:
How about defining it in the `CHIPCryptoPALmbedTLS.h` header? Then it could be reused in `CHIPCryptoPALPSA.cpp`, instead of defining it again. The same for `CHIP_CRYPTO_PAL_PRIVATE_X509`.